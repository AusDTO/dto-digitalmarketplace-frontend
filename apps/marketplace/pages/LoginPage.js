import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { withRouter, Redirect } from 'react-router-dom'

import BaseForm from 'shared/form/BaseForm'
import formProps from 'shared/form/formPropsSelector'

import LoginForm from '../components/LoginForm'
import { login } from '../actions/appActions'

export class LoginPageComponent extends BaseForm {
  static propTypes = {
    model: PropTypes.string.isRequired,
    handleSubmit: PropTypes.func.isRequired,
    loggedIn: PropTypes.bool.isRequired,
    currentlySending: PropTypes.bool.isRequired
  }

  constructor(props) {
    super(props)
    this.state = {
      submitClicked: null,
      currentlySending: null
    }
  }

  onSubmitClicked = () => {
    this.setState({
      submitClicked: new Date().valueOf()
    })
  }

  render() {
    const { model, loggedIn, handleSubmit, currentlySending } = this.props
    const { from } = this.props.location.state || { from: { pathname: '/' } }

    return (
      <div className="row">
        <div className="col-sm-push-2 col-sm-8 col-xs-12">
          <article role="main">
            {loggedIn
              ? <Redirect to={from} />
              : <LoginForm
                  submitClicked={this.onSubmitClicked}
                  handleSubmit={handleSubmit}
                  model={model}
                  currentlySending={currentlySending}
                />}
          </article>
        </div>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  ...formProps(state, 'loginForm'),
  loggedIn: state.app.loggedIn,
  currentlySending: state.app.currentlySending
})

const mapDispatchToProps = dispatch => ({
  handleSubmit: values => dispatch(login(values))
})

const LoginPage = withRouter(connect(mapStateToProps, mapDispatchToProps)(LoginPageComponent))

export default LoginPage
