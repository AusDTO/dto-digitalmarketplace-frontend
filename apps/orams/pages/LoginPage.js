import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { withRouter, Redirect } from 'react-router-dom'

import LoginForm from 'shared/Login/LoginForm'
import BaseForm from 'shared/form/BaseForm'
import formProps from 'shared/form/formPropsSelector'
import { login } from 'orams/actions/appActions'
import { rootPath } from 'orams/routes'

export class LoginPageComponent extends BaseForm {
  static propTypes = {
    model: PropTypes.string.isRequired,
    handleSubmit: PropTypes.func.isRequired,
    loggedIn: PropTypes.bool.isRequired,
    currentlySending: PropTypes.bool.isRequired
  }

  constructor(props) {
    super(props)
    this.state = {
      submitClicked: null,
      currentlySending: null
    }
  }

  onSubmitClicked = () => {
    this.setState({
      submitClicked: new Date().valueOf()
    })
  }

  render() {
    const { model, loggedIn, handleSubmit, currentlySending, userType } = this.props
    const initialPage = userType === 'buyer' ? '/orams/seller-catalogue' : '/orams/profile'

    return (
      <main>
        <div id="login-page">
          {loggedIn
            ? <Redirect to={initialPage} />
            : <LoginForm
                submitClicked={this.onSubmitClicked}
                handleSubmit={handleSubmit}
                model={model}
                currentlySending={currentlySending}
                rootPath={rootPath}
                framework="ORAMS"
              />}
        </div>
      </main>
    )
  }
}

const mapStateToProps = state => ({
  ...formProps(state, 'loginForm'),
  loggedIn: state.app.loggedIn,
  currentlySending: state.app.currentlySending,
  userType: state.app.userType
})

const mapDispatchToProps = dispatch => ({
  handleSubmit: values => dispatch(login(values))
})

const LoginPage = withRouter(connect(mapStateToProps, mapDispatchToProps)(LoginPageComponent))

export default LoginPage
