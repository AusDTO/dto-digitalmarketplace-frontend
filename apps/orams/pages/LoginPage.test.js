import React from 'react'
import expect from 'expect'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import { Provider } from 'react-redux'

import configureStore from 'orams/store'
import LoginPage from './LoginPage'

describe('Test suite for LoginPage component', () => {
  const state = {}

  const store = configureStore()
  test('LoginPage renders without errors', () => {
    const renderer = new ReactShallowRenderer()
    const tree = renderer.render(
      <Provider store={store}>
        <LoginPage {...state} />
      </Provider>
    )

    expect(tree).toMatchSnapshot()
  })
})
