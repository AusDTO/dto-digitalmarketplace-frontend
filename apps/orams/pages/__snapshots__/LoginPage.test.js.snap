// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for LoginPage component LoginPage renders without errors 1`] = `<withRouter(Connect(LoginPageComponent)) />`;
